packages/pages/src/dev/server/middleware/sendAppHTML.ts

```diff
@@ -52,10 +52,16 @@ export default async function sendAppHTML(
     ? templateModuleInternal.getHeadConfig(props)
     : undefined;
 
+  const hydrate =
+    templateModuleInternal.config.hydration === undefined
+      ? false
+      : templateModuleInternal.config.hydration;
+
   const clientHydrationString = getHydrationTemplateDev(
     clientServerRenderTemplates.clientRenderTemplatePath,
     templateModuleInternal.path,
-    props
+    props,
+    hydrate
```

Comment:
Once you update the internal type you can just do templateModuleInternal.config.hydration